ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/snippet/factory/BasicSnippetFactory.java

```diff
@@ -153,19 +155,38 @@ public List<VariableDeclarationSnippet> createVariableDeclarationSnippets(Node n
                 qualifiers = NodeFactory.createNodeList(varNode.finalKeyword().get());
             }
 
-            if (((VariableDeclarationNode) node).initializer().get().kind() == SyntaxKind.QUERY_ACTION) {
+            Optional<ExpressionNode> optVarInitNode = varNode.initializer();
+            if (optVarInitNode.isEmpty()) {
+                assert false : "This line is unreachable as the error is captured before.";
+                addErrorDiagnostic("Variable declaration without an initializer is not allowed");
+                return null;
+            }
+
+            ExpressionNode varInitNode = optVarInitNode.get();
+            SyntaxKind nodeKind = varInitNode.kind();
+            if (isSupportedAction(nodeKind)) {
                 TypedBindingPatternNode typedBindingPatternNode = varNode.typedBindingPattern();
                 TypeDescriptorNode typeDescriptorNode = typedBindingPatternNode.typeDescriptor();
                 BindingPatternNode bindingPatternNode = typedBindingPatternNode.bindingPattern();
-                String functionPart = "function() returns ";
-                String functionBody = varNode.initializer().get().toString();
-                String functionString = "var " + "f_" + varFunctionCount + " = " + functionPart
-                        + typeDescriptorNode.toString() + "{" + typeDescriptorNode + bindingPatternNode.toString()
-                        + " = " + functionBody + ";" + "return " + bindingPatternNode + ";};";
+
+                // Check if the type descriptor of the variable is 'var' as it is not yet supported.
+                if (typeDescriptorNode.kind() == SyntaxKind.VAR_TYPE_DESC) {
+                    addErrorDiagnostic("Actions not allowed with a 'var' TypeDescriptor");
```

Comment:
better to say "not yet supported" in the error than "not allowed".
HBT?
```suggestion
                    addErrorDiagnostic("'var' type is not yet supported for actions. Please specify the exact type.");
```